### PR TITLE
fix(shell): Shell UI QOL adjustments

### LINKED ISSUE #5983: [shell] it should not be possible to set a blank screen name

### LINKED ISSUE #6091: [shell] emoji keyboard off-screen on mobile ios
IMG_1066

### PR DESCRIPTION
**Pull request by thure**
Resolves #5983. Resolves #6091.

This PR:
- Only updates the profile with truthy values, preventing an empty string `displayName`
- Opens the emoji picker on top of the trigger by default, only opens alongside the trigger at `md` or greater.

https://github.com/dxos/dxos/assets/855039/7ec99cc0-6732-4055-b822-42e670510b47

### CODE REVIEW

File: packages/sdk/shell/src/panels/IdentityPanel/steps/ProfileForm.tsx [hunk 1]
```diff
@@ -106,7 +106,12 @@ const ProfileFormImpl = (props: ProfileFormImplProps) => {
         <Action
           variant='primary'
           disabled={disabled}
-          onClick={() => onUpdateProfile?.({ displayName, data: { emoji, hue } })}
+          onClick={() =>
+            onUpdateProfile?.({
+              ...(displayName && { displayName }),
+              ...((emoji || hue) && { data: { ...(emoji && { emoji }), ...(hue && { hue }) } }),
```
Comment (wittjosiah) [inline packages/sdk/shell/src/panels/IdentityPanel/steps/ProfileForm.tsx:112]: this is a lot to look at...but I get it 😅

**Review by wittjosiah — APPROVED** [review]

File: packages/sdk/shell/src/panels/IdentityPanel/steps/ProfileForm.tsx (hunk 1 above)
Comment (thure) [reply packages/sdk/shell/src/panels/IdentityPanel/steps/ProfileForm.tsx:112]: I was thinking similarly 🥲

Comment (ZaymonFC) [reply packages/sdk/shell/src/panels/IdentityPanel/steps/ProfileForm.tsx:112]:
```ts

// Helper type to extract keys that are in both objects
type GetObjSameKeys<T, U> = {
  [K in keyof T & keyof U]: (K extends keyof T ? T[K] : never) & (K extends keyof U ? U[K] : never);
};

// Helper type to extract keys that are unique to each object
type GetObjDifferentKeys<T, U> = Omit<T, keyof U> & Omit<U, keyof T>;

type MergeTwoObjects<
  T,
  U,
  T0 = Partial<GetObjDifferentKeys<T, U>> & { [K in keyof GetObjSameKeys<T, U>]: DeepMergeTwoTypes<T[K], U[K]> },
  T1 = { [K in keyof T0]: T0[K] },
> = T1;

export type DeepMergeTwoTypes<T, U> = [T, U] extends [object, object] ? MergeTwoObjects<T, U> : T | U;

const isObject = (item: any): item is Record<string, any> => item && typeof item === 'object' && !Array.isArray(item);

const deepMerge = <T extends Record<string, any>, U extends Record<string, any>>(
  obj1: T,
  obj2: U,
): DeepMergeTwoTypes<T, U> => {
  const result: Record<string, any> = { ...obj1 };
  Object.keys(obj2).forEach((key: string) => {
    if (isObject(obj2[key])) {
      if (isObject(result[key])) {
        // Recursively merge object values
        result[key] = deepMerge(result[key], obj2[key]);
      } else {
        // Assign directly if the key does not exist in result or is not an object
        result[key] = obj2[key];
      }
    } else {
      // Directly assign non-object values
      result[key] = obj2[key];
    }
  });
  return result as DeepMergeTwoTypes<T, U>;
};

const addIf = <V extends object>(condition: any, value: V): Partial<V> => (condition ? value : {});

const displayName = 'Name';
const emoji = '👋';
const hue = 200;

const profileUpdate = deepMerge(
  addIf(displayName, { displayName }),
  addIf(emoji || hue, { data: deepMerge(addIf(emoji, { emoji }), addIf(hue, { hue })) }),
);

// typeof profileUpdate =
// const profileUpdate: {
//    displayName?: string | undefined;
//    data?: {
//        emoji?: string | undefined;
//        hue?: number | undefined;
//    } | undefined;
// }
```

Well this was a fun puzzle for not much payoff. (Not suggesting we use this, but I thought it was fun).

Comment (ZaymonFC) [reply packages/sdk/shell/src/panels/IdentityPanel/steps/ProfileForm.tsx:112]: Typescript is cursed sometimes.

**Pull request merged**